llama_index/output_parsers/utils.py

```diff
@@ -14,7 +16,16 @@ def parse_json_markdown(text: str) -> Any:
     json_string = text.split("```json")[1].strip().strip("```").strip()
     try:
         json_obj = json.loads(json_string)
-    except json.JSONDecodeError as e:
-        raise OutputParserException(f"Got invalid JSON object. Error: {e}")
+    except json.JSONDecodeError as e_json:
+        try:
+            # NOTE: parsing again with pyyaml
+            #       pyyaml is less strict, and allows for trailing commas
+            #       right now we rely on this since guidance program generates 
+            #       trailing commas
+            json_obj = yaml.safe_load(json_string)
```

Comment:
Is there a reason why we don't just use this first?

Reply:
honestly pretty arbitrary choice. apparently json load is faster than yaml load.